docs/developer/architecture/centralized-cli.rst

```diff
@@ -0,0 +1,172 @@
+CLIs:
+
+- CLI follows kubectl style: kubectl <action> <resource>
+- CLI provides an exhaustive help, per action, with relevant examples
+- CLI provides <action> help when the command is not valid
+- CLI is not interactive (except if password input is needed)
+- CLI should not require password input
+- CLI provides a dryrun mode for intrusive operations
+- CLI provides a verbose (or debug) mode
+- CLI implementation relies on secure APIs
+- CLI support <action> completion for easy discovery
+- CLI output is standardized and human readable by default
+- CLI output can be formatted in JSON or YAML
+- CLI can be executed from outside of the cluster (except maybe for very first
+  install script)
+
+When it is possible, it would make sense to leverage kubectl plugin
+
+All functionalities are exposed through 2 distincts CLI:
+- kubectl (enriched using kubectl approach)
+- mk8sctl: a new CLI, exposing specific MetalK8s functionalities
+
+In order to operate the cluster with mk8sctl from outside of the cluster, a
+specific pkg (for each OS) or a procedure explaining how to deploy it is
+available.
+The mk8scli is deployed and available by default on bootstrap nodes
+
+Requirements
+------------
+
+Not listing all commands that are already available through kubectl.
+Only describing commands that are missing or commands that can be simplified
+using new command line arguments and predefined manifest.
+
+
+Cluster Resources Administration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**tool: kubectl**
+
++------------+------------------+-----------------------------------------+
+| Resource   | action           | parameters                              |
++============+==================+=========================================+
+| node       | add              | name, ssh-user, hostname or ip,         |
+|            |                  | ssh port, ssh-key-path, sudo-required,  |
+|            |                  | roles, manifest-template-path           |
++------------+------------------+-----------------------------------------+
+| the manifest-template-path is the path to the manifest describing the   |
+| resource. Everything that is in the command line, override the manifest |
+| content. This apply to Node and Volumes.                                |
++------------+------------------+-----------------------------------------+
+| node       | edit             | name, hostname (to change node hostname)|
++------------+------------------+-----------------------------------------+
+| node       | deploy           | <list of nodes>, dry-run                |
++------------+------------------+-----------------------------------------+
+| node       | delete, drain,   | <list of nodes>                         |
+|            | taint, label     |                                         |
++------------+------------------+-----------------------------------------+
+| volume     | add              | name, nodeName, storageClassName,       |
```

Comment:
This (`devicePath` and such) is too tied to the current type of volume we may support (and only one of the two we do support). This needs to be revised into a proper CLI UX.

Reply:
this one for CLI will be a very long command line with all the "optional values" for each type of volume. It's almost like writing the whole yaml file describing the volume in this case

Reply:
@xaf-scality @NicolasT What I was suggesting is to have manifest templates and being able to override some values in command line ... to be discussed during today's meeting

Reply:
will remove this concept of manifest template